### ISSUE TITLE
proto-Treasury system for ETC

### ISSUE THREAD

**Issue by bpmckenna** [input-output-hk/ECIPs#2]
The issue is the discussion for https://github.com/ethereumclassic/ECIPs/pull/349 which proposes a Proto Treasury for ETC

**Comment by TokenHash** [comment]
>>>copying my comment which I originally wrote in the wrong page<<<

The boss in the system is the database. This means no one can hold any veto or reserve any special powers. Therefore, giving miners the ability to veto decisions creates an imbalance in the improvement proposal process.

The treasury is already an imbalance as it gives power to the few dev teams and super-voters to decide the future of the database which contains the decentralized programs, money, property, and agreements, which are the "sacred cow" of the system.

Adding an additional level of complexity by introducing miner veto only makes the treasury system worse.

I don't agree with the treasury, but if it will be implemented it has to be as simple and straight forward as possible as it gives transparency and accountability in the system on the decision makers (devs and super-voters) so they can't dilute their responsibility by pointing to anonymous miners for delays or bad decisions.

If there is going to be a treasury it should be a simple: miner tax -> fund -> proposal -> voting -> implementation.

Any additional complexity, although it gives the impression of an illusory "checks and balances", actually diminishes it.

The treasury is not a technical proposal it's a social engineering proposal, so it must not be confused and obscured as technical.

The treasury has to be as straight forward and simple as possible, and SHOULD NOT give special powers to miners. If power is given to miners in addition to the prerogatives they already have in the system, then division of power between nodes, devs and miners is diminished: https://etherplan.com/2019/05/18/proof-of-work-has-division-of-power-proof-of-stake-does-not/7619/

A clean treasury at least makes clear who is accountable.

**Comment by TokenHash** [comment]
PS: I think this "issues" page for discussions about the treasury should be on the issues section of the EthereumClassic repo, where all discussions usually are.

cc @q9f

**Comment by TokenHash** [comment]
I would also eliminate the Gitcoin 10% distribution. I would only direct funds to the three client member firms ("client members" - MemberType.Client) who are known companies with a long participation in ETC (ETC Coop, IOHK, ETC Labs).

The Gitcoin idea is "decentralization" theatre and does not accomplish any goal (and decentralization in a treasury is actually a hazard, not a benefit). It will be an account with millions of dollars with stupid projects being proposed, with no control of fraud or weird results.

ETCDEV: As ETCDEV was suddenly kicked out by ETC Labs in 2018, and their main problem was lack of funding, I would invite https://github.com/splix to form a team and come back to work for ETC (perhaps Elaine Ou as well).

The above would make the distribution 25% to each of IOHK, ETC Coop, ETC Labs, and ETCDEV (if Igor agrees).

[to clarify, for the smartypants, I personally don't, would, and will not work for ETCDEV as I don't want to receive any funds from the Treasury for the reasons I wrote on my post: https://etherplan.com/2020/08/14/why-ethereum-classic-should-not-create-a-treasury/12384/]

Voting: I don't know what is the voting mechanism for if the client members will get the funding (30%x3 or 25%x4). The voting seems like a cumbersome process which will be full of political and governances problems. (And only the big guys would arbitrarily decide the projects).

**Comment by gitr0n1n** [comment]
> The community also has the power to add members to the p-ECTS. Only MemberType.Client can be added. There is no limit to the number of clients.


Request:
I would like to formally request a dev team named `Ethereum Core`, or `E.C.` for short, be added as a a `MemberType.Client`. Rough proposal: Upon receipt of the stable funding, we will fork a client that needs `material improvement` in order to acquire adoption, e.g. Besu.  `Adding Value` to the network by developing this client to be a competitive option in the Ethereum Classic marketplace for those that particiant in the PoW side of the project (miners).

Thanks! - r0n1n
[ethereumcore.org](https://ethereumcore.org)

Client Fork Repo: https://github.com/ethereumcore-org/besu

**Comment by erdnapa** [comment]
This proposal should be rejected without regard to its merit, as it violates ECIP 1017, Monetary Policy and Final Modification to the Ethereum Classic Emission Schedule.

An easy fix would be to make the p-ECTS payment mandatory, however, there is plenty to be said with regard to its merit as well (funding should be competitive, voting should have costs to show vested interest, etc).

**Comment by TokenHash** [comment]
**Final Comments of The Ethereum Classic Proto Treasury Proposal**

- The format provides 100% fixed funding upfront with no apparent accountability from client members or Gitcoin grantees.
- It does not make sense to send funds blindly to Gitcoin without knowing how they will be allocated, what they will finance, and with no control of what the grantees will do or even if they will do it. The Gitcoin member should be removed, imo.
- Terms as “wider community development”, “evolving development of the ETC platform”, “clear vision”, etc. are vague and blurry aspirational statements, but do not really provide any concrete requirement for plans or roadmaps from the client members.
- That funds are directed to client members and then their proposals go through the ECIP process is a statement that provides a false sense of community control over funds and projects. The reality is that once client members receive such large amounts of capital they will gain significant control of the network and its future development.
- Due to the above, and that voting is mapped 1 to 1 to ETC token units, also makes the “collective funding decisions” statement a false aspiration, as very few will decide future changes.
- There is no advancement nor installation of a real decentralized “governance” system. The treasury just establishes three client members as the new dirigents of ETC. The ECIP process, the treasury proposal process, and the supposed ability to add or remove client members become a sort of “decentralization theatre” because the client members, by themselves or together with large staking voters, will have overriding power over the roadmap and decision making.
- The “built in obsolescence” or “shut down” feature of the treasury is also a false sense of community control for the same reasons as above.
- Miner ability to burn funds gives unnecessary power to miners as expressed in comments I made before in this issue. Miners already have their role and power in the network so the miner tax burning feature should be removed, imo.
- The assumptions that client teams will provide updates and share decision making with the community also provides a false sense of community control. It seems incentives for client members are to actually shut down the community and implement their preferred projects.
- The only real balance of power this proto-treasury establishes is whatever adversarial relationship the client members may have between themselves, and the normal adversarial relationship between client members, miners, and node operators. 
- In fact, the relationship between client members and miners and node operators will be similar to a provider-customer relationship. Very much how a large tech company provides services to enterprise customers e.g. Red Hat. But paid by the network, not the customer.
- As this proto-treasury pushes funds to client members and Gitcoin a priori, there is no incentive for them to actually propose innovation, or efficient servicing of the network. As miners do today, they can just sell the ETC they get for USD, BTC, or ETH, and just continue acting as if they were interested in ETC. Or, they will keep enough funds and rich ETC owner allies to favorably vote for every proposal on the treasury.  
- For all of the above there will likely be no control over project lifecycles, timing, quality, nor any other assurance parameter.
- In the case of the Gitcoin member, it is even worse because the separation and layers between treasury or community and the final grantees are even more, and there is no control of delivery or assurances. Especially for external marketing and other difficult to monitor and measure activities.

**Questions about some details:**

- The assumptions mention “smart contracts” in the following way: “All smart contracts will go through one or more professional security audits before deployment. After deployment, independent parties will need to verify that the deployed contract corresponds exactly to the audited version. All ETC clients will include this in their actual hard fork proposal.” What are smart contracts in this context? is it the treasury smart contract itself?
- The assumptions also mention “supply changes will require a hard fork”, does this “supply” refer to the 20% miner tax?
- The voting mentions a 60% threshold of yes votes over no votes, does this mean that a simple majority of 60% of votes is needed to effect changes to the treasury? Or, is there a different calculation for this voting mechanism?
- The “quorum” is set at “1/3 of total ETC supply”, what does “supply” mean here? The total ETC supply of 117 million ETC? The stock of ETC in the treasury itself? Or the staking ETC in each proposal? Other?

**Suggestions:**

- Remove Gitcoin member
- Remove miner burn feature
- Change funding of client members from a priori funding, to a posteriori funding by ECIP proposal. This transfers some real power back to “the community” and gives a real function to the ECIP process. 
- If client members will not be incentivized to work for ETC if the funding will be conditioned by the politics of the ECIP process, then a portion of their allocation could be guaranteed to pay for fixed costs, and then their ECIP proposals could be further funded on a case by case basis as in the previous point. For example, the treasury could send 50% of the miner tax to client members (1/3 each), and the other 50% could be retained in the treasury to be paid to them only on an ECIP by ECIP basis. This would also provide some control by the community and create more accountability from the art of the three client members.
- Put a definite termination term for the proto treasury where it will shut down automatically and the community has to hard fork to actually continue it. This is a stronger guarantee than having to vote to shut it down.

**Comment by TokenHash** [comment]
I have two more comments about the risks of the Treasury I wanted to share, that I think illustrate how careful it has to be designed:

**Control:**

- Client members will receive millions of ETC in 15 years (see cash flow below)
- ETC tokens are used 1:1 for voting proposals (which eventually will possibly include ECIP funding proposals as well)
- Rich friends (like Grayscale who has custody of and fiduciary management of ~10 million ETC) may become -or very likely will- ally with client members
- Client members actually physically control the client node software
- In other words, will have the money and the votes, therefore the unilateral implementation and deployment power of changes to ETC

**Cash flow:**

As I said before, the pool of capital and miner tax are not the main problem (aside from the reduced security budget), the problem is the decision making centralization.

If you think about it, the future cash flow of the Treasury will be 16.4 million ETC (200-118=82; 92*20%=16.4 million), the majority of which occurs in the next 15 years.

The treasury will get approximately 12.8 million ETC from today to 2036 (15 years). And at today’s price ($8) that is $102 million.

If ETC is successful (perhaps b/c the Treasury creates a sort of good will in the market, accelerates innovation, and changes like keccak256, flyclient, etc. will be welcome as well) then that could easily be $1 billion, or several $$$ billion....

I do not think it is wise to send that money blindly to three fixed entities (this is not a doubt about the proposed client members, but just basic risk management, any client member, however trusted, would represent the same security hazard).

The above means that even more definitely the Treasury has to be ended by default in 5 years.

One simple way would be to set it at 20% tax for 5 years (or 10,512,000 blocks) and 0% from then on by default. Then, if the ecosystem wants to increase it again it has to hard fork again.

**Comment by TokenHash** [comment]
I just noticed ECIP-1098 does not have a license. Kindly add a license to the ECIP. Recommended licenses in Ethereum Classic improvement proposals are:

Recommended licenses:
Apache-2.0: Apache License, version 2.0
BSD-2-Clause: OSI-approved BSD 2-clause license
BSD-3-Clause: OSI-approved BSD 3-clause license
CC0-1.0: Creative Commons CC0 1.0 Universal
GNU-All-Permissive: GNU All-Permissive License

Source: https://github.com/ethereumclassic/ECIPs/blob/master/_specs/ecip-1098.md

**Comment by TokenHash** [comment]
Inline with everything I have commented and proposed above, I have proposed a replacement for the Gitcoin portion of the proto treasury system. This ECIP is marked as complementary to ECIP-1098:

ECIP-1102: Use The Etherplan Consensus Engine for Approving Proto Treasury Funding Proposals:

https://github.com/ethereumclassic/ECIPs/pull/408

**Comment by nogo10** [comment]
"Votes will be counted as total locked ETC that has participated (voted yes/no) on each proposal." 
If I read it correctly, the votes one person can cast is based primarily on the number of ETC they stake?
I think it is feasible and more robust to allow one per participant whether they be developers, miners or large ETC holders. All have equal sweat equity in the success of ETC.. Technically an open source captcha could be used to dissuade Sybil attacks. The upside is a higher of participants.. having quorum is easier to achieve if all participants voice has a chance to be heard with voting rather than drowned out by a whale. 

TLDR: a developer who spent thousands of hours of his time should have equal say to a large holder essentially 'buying 'Votes'

**Comment by nogo10** [comment]
An anti Sybil protocol could be implemented with captcha bots on the Discord discussion channel.. 

Another broadband approach is to use a tipping bot on Discord or Reddit to issue voting tokens based on much more than ETC balance: a participating attribution, if you will:
 upvotes
 tipping bot
 etc balance
 channel seniority
 number of deployed dapps
 membership level 
 referrals
 number of transactions

**Comment by TokenHash** [comment]
@nogo10 I don't particularly believe in the fairness of having investors and developers having equal votes, and there may be investors with 1 to 10 million, and developers who contribute one to 1 million loc or commits. Also, there are many non-investor and non-developer participants like node operators, community managers, miners, etc. who are weak or strong contributors. Not only are these differences everywhere, but the intensity of participation and their influence changes all the time as their interest and time in ETC varies over time.

For all of the above, either having staker vote or 1:1 proof of authority (the way to give single votes to deserving participants) votes to active participants are both flawed mechanisms. I would say there is no mechanism that may capture the reality of importance, influence, stake, and participation (and whatever other variables there may be) of the participants. The only mechanism, which is not a mechanism per se, is "rough consensus", which is basically how ETC works now.

What I propose is to have a similar system as the ECIP process, which is a rough consensus process, but for funding proposals. This system may be "naked" as the ECIP process is, where there is no voting or decision making protocol, just people agreeing on proposals or not. This system could be called the Ethereum Classic Funding Proposal (ECFP) process, and it could be managed on a separate repo, like `/github/ethereumclassic/ECFPs`.

The other thing that may be added, but would start to resemble a flawed voting mechanism, is a sort of "rough consensus" imitating mechanism on top. I proposed this in this PR https://github.com/ethereumclassic/ECIPs/pull/408.

**Comment by nogo10** [comment]
Rough consensus in a GitHub channel brings open discussion and better  transparency  to the process and record keeping as well. Those are advantages. The 'show of hands ',  so easy to do in meatspace, needs to happen to have some closure and finality to the process to move things forward. GitHub is asynchronous and scheduling problems can hinder participants. So instead use approval upvotes/survey to keep a running tab on current options with a well published timeline. Even something as simple as a doodle poll would be effective.

So in summary: participation open to all via comments + upvotes survey polls on a fixed timeline with issue tracking.

Disclosure: I've been working on a decentralized solution that can solve this type of problem. Will be posting alpha in the coming months.

**Comment by nogo10** [comment]
> difficult to associate proposals as to which are competing and which are complementary.

That is more difficult to solve, I guess a member is assigned to moderate topics/process in discussion channels. Such a member would not be able to vote but can assign addresses to a votinglist. I believe Discord plugins exist for that purpose.

**Comment by TokenHash** [comment]
Yes, the system I proposed has a PR and PR merge process with administrators to do that, very similar to how editors analyze and merge PR in the ECIP process. Then the proposals go the the voting mechanism, which I think resembles rough consensus.

**Comment by TokenHash** [comment]
I suggest not selecting ETC Labs or any of their subsidiaries or employees as one of the "client members" - MemberType.Client.

**Comment by q9f** [comment]
Will be difficult to rally support behind such a proposal if you desire to exclude parties right from the beginning.

**Comment by TokenHash** [comment]
The treasury is not a fountain of charity money to give it to the world or the 30 million developers on GitHub. It is not a blockchain per se either. It selects specific teams who will work for the system on the basis of their qualifications. It is by definition an exclusion system. My proposal is to not select an entity, subsidiary, and active employees who are not desired by me in particular to have any function in ETC as they have disqualified themselves. This ECIP-1098 is a security hole for ETC and has to be very carefully implemented (and limited in time).

Ex-employees who have no interests or are completely delinked from that entity could be candidates, but hijacking community resources and centralizing the system shows misjudgment and is an ethical failure in the context of pow blockchains, imo.

What I think in particular is only one opinion, not rough consensus.

**Comment by TokenHash** [comment]
I would say anyone who is completely delinked from that entity would be fine for me...just my opinion.

**Comment by erdnapa** [comment]
I've mentioned this elsewhere, but will include it here as part if the official discussion:

Two main issues:

1. The treasury should NEVER be optional. Either everyone contributes  or nobody does. Having the option to sending etc to a burn address turns the treasury into a donation model - the exact thing the treasury is supposed to replace.
2. The treasury should NOT have a fixed distribution, so no teams, individual developer ratios etc. Those lead to laziness , since teams will get paid no matter what.

A better treasury would fund submitted proposals based the the merit as judged by votes. I'll outline a 'back of the napkin' treasury system, that would fix some of the shortcomings of the current ECIP:

* 10% mandatory treasure fee per block
* anyone can submit proposals, as many as they want
* voting and funding every 6 months at a predetermined block number
* Fund distribution exactly 1 month later
* Proposal application fee is 1% of requested funds
* Maximum funding per proposal 150,000 ETC (a bit less than 50% of funds), no minimum funding level
* Voting:- each YES/NO vote costs 0.1ETC per address to vote for the entire balance of the address (=shares)- any ETC address can change its vote as often as it wants, but only the last vote before the voting block counts. There is no refund of the earlier 0.1 ETC vote fee.
* Funding:- if more ETC are requested than are available: the funding will be reduced at the same percentage for all proposals, down to 75% of requested funds- the proposals with the highest YES/NO ratio get funded in order, until a project will receive less than 75% of its request, this and all other proposals with lower YES/NO ratio will receive no funding
* Result Evaluation:- Funded proposals have to submit Software/Results/Progress Report... 10 months after funding.- If further development is possible or necessary: Projects can ask for renewal/extension of funding at the same or lower level than the original proposal for a new funding round without application fee (up to 2x).- Renewals go into the proposal pool and are voted on the same as other proposals

The system outlined above has several advantages over the current proposal: it assures that the proposals with the highest merit get funded, it eliminates fixed ratios to teams and rather funds those based on the quality of their proposals and work. The system assures that voters are vested in the outcome of the vote by charging a small but noticeable fee. Finally, application and vote fees ensure that the treasury itself has enough funding to ensure the process runs smoothly.

Also note: such a treasury system is not necessary limited to development only, proposals could be submitted for any kind of project. For example: new client/wallet/blockchain dev, running a HA rpc service for the community, sending Etc into space, gifting each crypto journalist that wrote about ETC an ETC ornament..... But the community will vote and decide what it deems necessary.

Tweaking and optimization are surely required.

**Comment by IstoraMandiri** [comment]
https://www.ethereumclassicclassic.org/
